### PR TITLE
feat(rosetta-rpc): Add Metadata tagging fee-related Operations

### LINKED ISSUE #7654: Rosetta: Add Metadata Tagging Fee Related Operations
Today, a standard transfer ([explorer example](https://explorer.near.org/transactions/Ai8LX3GQEdSrFSRpyKqVvFQyroX1ycFEUc8ZhDwMjMQh)) will have 3 parts:

1.  `tx:Ai8LX3GQEdSrFSRpyKqVvFQyroX1ycFEUc8ZhDwMjMQh`, 2 payment operations. `3.3538 NEAR` as the transfer and `0.00008618 NEAR` as the gas prepayment.
2.  `receipt:6DEuh78kRc4TM2bFuuRTjAzAfYnidfZV1nh4zaozRA6t`, 1 payment operation, `3.3538 NEAR`
3.  `receipt:AYwML97wNXKSbNTqhPRRrYEv3bHuVXpuaM4Bw3heHU1Q`, 1 gas refund operation, `0.00000127 NEAR`

We need to be able to differentiate the `0.00008618 NEAR` gas prepayment operation from the `3.3538 NEAR` operation in the initial tx. Ideally we can also mark the gas refund operation.

 curl -H "Content-Type: application/json" -d '{ "network_identifier": 

 {"blockchain":"nearprotocol", "network": "mainnet"} 

, "block_identifier": 

 {"index":74553709} 

 }' 'localhost:3000/block'

 RESPONSE:

 {
 "transaction_identifier": 

 { "hash": "tx:Ai8LX3GQEdSrFSRpyKqVvFQyroX1ycFEUc8ZhDwMjMQh" } 

,
 "operations": [
 {
 "operation_identifier": 

 { "index": 0 } 

,
 "type": "TRANSFER",
 "status": "SUCCESS",
 "account": 

 { "address": "25ab96ddd0d9a8e0ab48b45c2ef7473a83c4ff7e49239b46613b3d688e024731" } 

,
 "amount": {
 "value": "-3353895615031144000000000",
 "currency": 

 { "symbol": "NEAR", "decimals": 24 } 

 }
 },
 {
 "operation_identifier": 

 { "index": 1 } 

,
 "type": "TRANSFER",
 "status": "SUCCESS",
 "account": 

 { "address": "25ab96ddd0d9a8e0ab48b45c2ef7473a83c4ff7e49239b46613b3d688e024731" } 

,
 "amount": {
 "value": "-86184677687500000000",
 "currency": 

 { "symbol": "NEAR", "decimals": 24 } 

 }
 }
 ],
 "related_transactions": [
 {
 "transaction_identifier": 

 { "hash": "receipt:6DEuh78kRc4TM2bFuuRTjAzAfYnidfZV1nh4zaozRA6t" } 

,
 "direction": "forward"
 }
 ],
 "metadata": 

 { "type": "TRANSACTION" } 

 },

Ideally all gas payments (and gas refund) can be tagged with metadata (on the operation, not the transaction). Previously `predecessor_id.address = 'system'` was used to identify gas refunds, but it doesn't seem to be working anymore. Maybe just having `metadata.operation_type = 'fee'` and `metadata.operation_type = 'gas_refund'` could work? Or just tag both as `fee`?

 "metadata" : {
 "predecessor_id" : 

 { "address" : "system" } 

 },

### PR DESCRIPTION
**Pull request by frol**
Closes #7654.

This PR adds `metadata.transfer_fee_type` field to the Operation object, which is only set for Gas prepayment (which happens on transaction processing) and Gas refunds (receipt that is sent by `system` user). Thus, there are just three states:

* not set - it is either not a transfer operation (Construction API), or it is not a fee transfer
* `GAS_PREPAYMENT`
* `GAS_REFUND`

### CODE REVIEW

**Comment by matklad** [comment]
(note firat is OoO, so this'll probably wait until next week for review, unless there's a strong need otherwise)

**Comment by frol** [comment]
@matklad The implementation was verified by the issue author: https://github.com/near/nearcore/issues/7654#issuecomment-1261081556

Can someone review it on the high-level / code quality, so we can merge it sooner rather later?

**Comment by frol** [comment]
Is there a chance to get it included in the 1.30.0 release?

File: chain/rosetta-rpc/src/types.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ use std::fmt;
     serde::Deserialize,
 )]
 #[serde(transparent)]
-pub struct AccountId(pub(crate) near_primitives::types::AccountId);
```
Comment (matklad) [inline chain/rosetta-rpc/src/types.rs:19]: :+1:

**Review by matklad — APPROVED** [review]
LGTM!

This should get to 1.30, yeah

**Pull request merged**